internal/eval/eval.go

```diff
@@ -915,7 +914,16 @@ func evalJSONDEL(args []string, store *dstore.Store) []byte {
 		return diceerrors.NewErrWithMessage("invalid JSONPath")
 	}
 	results := expr.Get(jsonData)
-	err = expr.Del(jsonData)
+
+	hasBrackets := strings.Contains(path, "[") && strings.Contains(path, "]")
```

Comment:
Please add few cases in test where the key itself has `[`/`]`.
e.g: `JSON.SET data $ '{"key[0]": "value", "array[1]": [10, 20, 30]}'`.

We can validate the behaviour similar to redis.
Like below commands should work
e.g: `JSON.DEL data '["key[0]"]'` etc.

Reply:
@lucifercr07 I checked and Redis doesn't allow square brackets in the key itself.

Reply:
@Dhroov7 works for me
```
127.0.0.1:6379> JSON.SET data $ '{"key[0]": "value", "array[1]": [10, 20, 30]}'
OK
127.0.0.1:6379> JSON.DEL data '["key[0]"]'
(integer) 1
127.0.0.1:6379> JSON.GET data
"{\"array[1]\":[10,20,30]}"
```

Reply:
@lucifercr07 Added the test case, please review